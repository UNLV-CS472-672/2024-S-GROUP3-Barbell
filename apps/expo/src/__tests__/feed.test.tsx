import React from 'react'

import { render } from '@testing-library/react-native'

import Activity from '~/components/feed/activity'

describe('Activity Component', () => {
  it('renders correctly with given props', () => {
    const mockUser = { name: 'John Doe' }
    const mockWorkout = {
      name: 'Morning Routine',
      description: 'A quick workout to get the day started',
      exercises: [{ name: 'Push-ups' }, { name: 'Sit-ups' }],
    }
    const mockWorkoutLog = {
      createdAt: new Date('2023-04-21'),
    }

    const { toJSON } = render(
      <Activity user={mockUser} workout={mockWorkout} workoutLog={mockWorkoutLog} />,
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
