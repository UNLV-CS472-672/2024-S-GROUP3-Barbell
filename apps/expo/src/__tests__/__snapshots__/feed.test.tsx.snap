// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Component renders correctly with given props 1`] = `
<View
  className="text-bb-slate-100 mx-3 p-3"
  style={
    {
      "borderBottomColor": "#737272",
      "borderBottomWidth": 1,
    }
  }
>
  <View
    className="flex-row items-baseline justify-between"
  >
    <Text
      className="mb-1 text-2xl font-bold text-slate-200"
    >
      John Doe
    </Text>
    <Text
      className="text-lg text-slate-200"
    >
      4/20/2023
    </Text>
  </View>
  <View
    className="flex-row items-baseline justify-between"
  >
    <Text
      className="m-2 text-lg text-slate-200"
    >
      Morning Routine
    </Text>
  </View>
  <Text
    className="m-2 text-slate-200"
  >
    A quick workout to get the day started
  </Text>
  <View
    className="my-3 flex-row"
  >
    <View
      className="flex-1"
    >
      <Text
        className="text-slate-200"
      >
        Exercises:
      </Text>
      <RCTScrollView
        data={
          [
            {
              "name": "Push-ups",
            },
            {
              "name": "Sit-ups",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={0.0001}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={null}
          >
            <Text
              className="mx-3 text-slate-200"
            >
              Push-ups
            </Text>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={null}
          >
            <Text
              className="mx-3 text-slate-200"
            >
              Sit-ups
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      className="flex-row items-end justify-items-end"
    >
      <Text />
    </View>
  </View>
</View>
`;
